src/eddymotion/model/_dipy.py

```diff
@@ -176,32 +114,30 @@ def __init__(
         """
 
         ReconstModel.__init__(self, None)
-        self.kernel = (
-            PairwiseOrientationKernel(
-                weighting=kernel_model,
-                a=a,
-                lambda_s=lambda_s,
-                sigma_sq=sigma_sq,
-            )
-            if kernel_model != "test"
-            else DotProduct() + WhiteKernel()
+
+        self.sigma_sq = sigma_sq
+
+        KernelType = SphericalKriging if kernel_model == "spherical" else ExponentialKriging
+        self.kernel = KernelType(
+            a=a,
+            lambda_s=lambda_s,
         )
 
     def fit(
         self,
         data: np.ndarray,
-        gtab: GradientTable | None = None,
+        gtab: GradientTable | np.ndarray,
```

Comment:
With the separation of `_dipy` and `_sklearn`, these types use GradientTables and arrays. Under `_sklearn` only arrays are allowed.